packages/storage/src/platform/browser/connection.ts

```diff
@@ -70,7 +70,7 @@ abstract class XhrConnection<T extends ConnectionType>
     if (this.sent_) {
       throw internalError('cannot .send() more than once');
     }
-    if (isCloudWorkstation(url) && isUsingEmulator) {
+    if (isCloudWorkstation(new URL(url).hostname) && isUsingEmulator) {
```

Comment:
This causes a unit test failure in "XhrConnection.send() should not reject on network errors"

That test passes a string called "testurl" in the `url` argument to `send` which `URL()` throws when receiving here because it is not a valid url